homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -432,6 +439,29 @@ def _get_beolink_jid(self, entity_id: str) -> str:
         # Return JID
         return cast(str, config_entry.data[CONF_BEOLINK_JID])
 
+    async def _async_update_sound_modes(
+        self, active_sound_mode: ListeningModeProps | ListeningModeRef | None = None
+    ) -> None:
+        """Update the available sound modes."""
+        sound_modes = await self._client.get_listening_mode_set()
+
+        if active_sound_mode is None:
+            active_sound_mode = await self._client.get_active_listening_mode()
+
+        # Add the key to make the labels unique (As labels are not required to be unique on B&O devices)
+        for sound_mode in sound_modes:
+            label = f"{sound_mode.name} ({sound_mode.id})"
+
+            self._sound_modes[label] = sound_mode.id
+
+            if sound_mode.id == active_sound_mode.id:
+                self._attr_sound_mode = label
+
+        # Set available options
+        self._attr_sound_mode_list = list(self._sound_modes.keys())
```

Comment:
We can remove `.keys()`. Copying a dict to a list copies the keys by default.

Reply:
Got it. I've created a PR to fix this https://github.com/home-assistant/core/pull/126762